syntax = "proto3";

package seq2seq;

message PromptRequest {
  string prompt = 1;
  // string userid = 2;
  // string orderid = 3;
}

message PromptReply {
  string message = 1;
  // string userid = 2;
  // string orderid = 3;
}

service Seq2Seq {
  // A simple RPC
  // rpc UnaryUnaryGreeting (PromptRequest) returns (PromptReply) {}

  // A response streaming RPC
  rpc PromptModel (PromptRequest) returns (stream PromptReply) {}

  // A request streaming RPC.
  // rpc StreamUnaryGreeting (stream PromptRequest) returns (PromptReply) {}

  // A bidirectional streaming RPC
  // rpc StreamStreamGreeting (stream PromptRequest) returns (stream PromptReply) {}
}